instat/dlgNewDataFrame.vb

```diff
@@ -52,45 +59,124 @@ Public Class dlgNewDataFrame
         ucrNewDFName.SetSaveTypeAsDataFrame()
         ucrNewDFName.SetLabelText("New Data Frame Name:")
         ucrNewDFName.SetPrefix("data")
-    End Sub
 
-    ' updating controls doesn't update the function
-    ' Sheet name is not on the dialog.
+        'ucrRdoOptions
+        ucrPnlDataFrame.AddRadioButton(rdoConstruct)
+        ucrPnlDataFrame.AddRadioButton(rdoCommand)
+        ucrPnlDataFrame.AddRadioButton(rdoRandom)
+        ucrPnlDataFrame.AddRadioButton(rdoEmpty)
+
+        'TODO:Providing conditions here may not be easy, find a way to do this properly!
+        ucrPnlDataFrame.AddFunctionNamesCondition(rdoConstruct, "data.frame")
+        ucrPnlDataFrame.AddFunctionNamesCondition(rdoCommand, ucrInputCommand.GetText())
+        'ucrPnlDataFrame.AddFunctionNamesCondition(rdoRandom, "")
+        'ucrPnlDataFrame.AddFunctionNamesCondition(rdoEmpty, "")
 
-    Private Sub ReopenDialog()
+        ucrPnlDataFrame.AddToLinkedControls(ucrNudCols, {rdoEmpty}, bNewLinkedAddRemoveParameter:=True, bNewLinkedHideIfParameterMissing:=True, bNewLinkedChangeToDefaultState:=True, objNewDefaultState:=2)
+        ucrPnlDataFrame.AddToLinkedControls(ucrNudRows, {rdoEmpty}, bNewLinkedAddRemoveParameter:=True, bNewLinkedHideIfParameterMissing:=True, bNewLinkedChangeToDefaultState:=True, objNewDefaultState:=10)
+        ucrPnlDataFrame.AddToLinkedControls(ucrInputCommand, {rdoCommand}, bNewLinkedAddRemoveParameter:=True, bNewLinkedHideIfParameterMissing:=True)
+        ucrNudRows.SetLinkedDisplayControl(lblRows)
+        ucrNudCols.SetLinkedDisplayControl(lblColumns)
+        ucrInputCommand.SetLinkedDisplayControl(lstLinkedControls)

```

Comment:
@shadrackkibet I have replaced this line (  `ucrInputCommand.SetLinkedDisplayControl(lstLinkedControls)` ) with 
` ucrInputCommand.SetLinkedDisplayControl(New List(Of Control)({lblCommand, btnExample}))` . I just thought it best to simplify(creating anonymous list out of array elements) this since the list is being used only once.